add in Run2.2i dr6d catalogs

heather999 opened the pull request:
Fix #482
This PR is adding the Run2.2i DR6d configuration files for the object and metacal catalogs.  
Need to also update the README on this repo to add in dr6d.

heather999 commented:
I see the output from GH Actions, which is [failing on pylint](https://github.com/LSSTDESC/gcr-catalogs/runs/1064126574?check_suite_focus=true). That's not a file I updated, but it would be nice to sort that out.
```
GCRCatalogs/dc2_object.py:642:0: E0012: Bad option value 'C0330' (bad-option-value)
```

yymao commented:
@heather999 thanks for the PR! Ccan you rebase onto (or merge with) master branch to fix the linter issue? 

Also, do we not need to wait for 4850?

heather999 commented:
Hi @yymao I just rebased the branch with master and a new GH action started up and seems to have completed successfully.
We're all set with 4850 - there were 12 patches that Eli identified we should exclude.  These have been removed from the pixel-level data and then I reran the make_object_catalog script to update the catalog for 4850.  I also updated the metacal output from 4850, to remove those same 12 patches.  So I think that's all set.

yymao commented:
@heather999 thanks Heather!  Let me do a double check to make sure the entries in metacal and object match and then will approve!

yymao commented:
Hmm, so 5 tracts that are available in object are not in metacal: 3075, 4034, 4431, 4639, 4850

And 9 tracts do not have matching entry number between object and metacal:

```
Tract  Object Metacal
----- ------- -------
2902   961671 1049842
2907  1016408  991492
2908   863574  918543
3080  1072869 1089049
4860    37480  246572
5065   313636  942335
5068   839000  863090
5069   796381  863919
5071   573270  862100
```

Should we exclude these 14 tracts in the composite catalog? 

cc @johannct

heather999 commented:
I'm confused why metacal data isn't available for 4850 - I see the file here: `/global/cfs/cdirs/lsst/shared/DC2-prod/Run2.2i/dpdd/Run2.2i-dr6d/metacal_table_summary/metacal_tract_4850.parquet` is it in some way incorrect?  
I also see metacal parquet files for 3075, 4034, 4431, and 4639 in the same directory.

yymao commented:
BTW, can you remind me --- is dr6d the final release of the WFD of DR6?

yymao commented:
The test passes! All metacal and object entries match.

heather999 commented:
Ah that's good news!
As for whether this is the final release of WFD DR6, it's not clear. Johann is continuing to process the DDF tracts and there is a real possibility there will be another release... But not sure yet.

yymao commented:
The README (I just updated it) still says "an early release of DR6" --- should we change the wording?

heather999 commented:
Maybe we could say
A near complete release of WFD Run2.2i DR6 excluding the tracts and patches in the DDF.

We could include a link to confluence where this should be further explained - planning to get that updated today https://confluence.slac.stanford.edu/display/LSSTDESC/DESC+DRP+DC2+Run2.2i+Processing

katrinheitmann commented:
I thought we did agree to call Eli's selection DR6-WFD. If all the DDFs are being processed, then that would be great and can be called something else. So as far as I understand this is "the DR6-WFD release" and the README should be updated. Thanks!



On 9/3/20 7:07 AM, Yao-Yuan Mao wrote:

The README (I just updated it) still says "an early release of DR6" --- should we change the wording?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/LSSTDESC/gcr-catalogs/pull/483#issuecomment-686444366>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCD3DGHZ5MVCU6IDTKJ5SLSD6BI5ANCNFSM4QT47O2Q>.

katrinheitmann commented:
We even have a github issue call DR6-WFD: https://github.com/LSSTDESC/DC2-production/issues/397

yymao commented:
Thanks @katrinheitmann --- I agree that dr6-wfd is a better name.  

@heather999 can you confirm that this dr6d entry is indeed the final dr6-wfd (i.e., not a preview version)?  And if so, we should just rename the configs and readme (and in this case, there will be no dr6d, just dr6-wfd).

heather999 commented:
Sure, let's rename everything, including the directory on Cori to dr6-wfd. I'll be back in a couple of hours to help with that.

johannct commented:
I am worried about a data product, any data product, that does not have a version attached to it. It is ok if the version is not in the name, but it needs to be unambiguous and predictive.

heather999 commented:
Understood and I agree, but also feel versioning should be in the name.. let's go with dr6-wfd-v1
I'll update the directory on Cori and then look at updating the config in this PR

heather999 commented:
@Yao does this look ok to you?

yymao commented:
it's fine but I wonder if we need to alias `dc2_object_run2.2i_dr6_wfd` to `dc2_object_run2.2i_dr6_wfd_v1`? We have not been using `_v1` as a version designation so I wonder if it's going to cause more confusion than clarify. I think it's ok to use v1 in the directory name but wonder if it needs to be user-facing?

johannct commented:
So can we step back for a little while to gather our thoughts properly? I agree with @yymao that we may want to provide the users with a transparent access to the "greatest latest' release, assuming that there may be more than one. But the user also needs to know what he/she is loading, so I would tend to feel better if gcr provided an info mechanism that outputs some basic information about the catalogue being pulled over by the user.... It could be something as simple as "loading release 1 of DC2 DR6 WFD data set" or something along these lines. GCR would need ideally to pull the information from the directory itself, which means that the shared root dir of a release should have a parseable file that provides data set name, release version, and a certain amount of provenance information. Now that we are about to officially release a final product that may be final, we should build some provenance info into the physical data set. Anyway, I think that it is not advisable to postpone this discussion any longer.... @wmwv @katrinheitmann @heather999 @jchiang87 inputs welcome.... We may need to move this to a larger audience

heather999 commented:
@yymao I'm certainly open to changing it 

@johannct I think we can take steps in the direction you are suggesting.  The description field of the catalog already states: `DC2 Run 2.2i DR6 WFD v1 Object Table` if we wish to modify that further that's a possibility. We can drop a parseable README into the catalog data directories as a starting point, where we can include additional details such as the software versions used, processing location, date, links to pertinent GH issues, etc - but I'm not necessarily suggesting we hold up this release of DR6 WFD for feature modifications of GCRCatalogs to actually parse that file at this time.

yymao commented:
@johannct we had these discussions when we released protoDC2/cosmoDC2. The "alias" feature is designed so that people can load these catalogs and automatically get their latest versions. Users can still specify the exact version they want to use. 

We also have implemented file size or md5 check for protoDC2/cosmoDC2, and the computed size/md5 are checked in to this repo to ensure the underlying files would not changed after the release. The downside is of course the checks make load time longer. 

If we don't like these implementations, we can of course brianstorm new ways. I just want to point out that we do have these means that we already applied to protoDC2/cosmoDC2.

heather999 commented:
@yymao What would we have to do to utilize the alias feature to alias `run2.2i-coadd-wfd-dr6` to `run2.2i-coadd-wfd-dr6-v1`?  

I will add yaml file to the `/global/cfs/cdirs/lsst/shared/DC2-prod/Run2.2i/dpdd/Run2.2i-dr6-wfd-v1` directory that will include some amount of provenance (date, processing info, software versions) - and we can plan to discuss this further within the data access group, our data czar and interested parties.

Once accepted, I would also update the existing symlink `/global/cfs/cdirs/lsst/shared/DC2-prod/Run2.2i/dpdd/Run2.2i-dr6` to point to `Run2.2i-dr6-wfd-v1` rather than `Run2.2i-dr6a` as it does now.

yymao commented:
Just needs a new config `run2.2i-coadd-wfd-dr6.yaml` with the following content: 

```yaml
alias: run2.2i-coadd-wfd-dr6-v1
```

yymao commented on README.md:
```diff
@@ -79,8 +79,8 @@ Confluence page (*DESC member only*).
 
-- `dc2_object_run2.2i_dr6c`: static object catalog for Run 2.2i DR6c (an early release of DR6; 165 tracts)
-- `dc2_object_run2.2i_dr6c_with_metacal`: static object + metacal catalog for Run 2.2i DR6c (an early release of DR6; 159 tracts)
+- `dc2_object_run2.2i_dr6_wfd_v1`: static object catalog for Run 2.2i DR6 WFD v1 (166 tracts)
+- `dc2_object_run2.2i_dr6_wfd_v1_with_metacal`: static object + metacal catalog for Run 2.2i DR6 WFD v1 (166 tracts)
```
Use the names without `v1` for the main user-facing entry?

```suggestion
- `dc2_object_run2.2i_dr6_wfd`: static object catalog for Run 2.2i DR6 WFD (v1; 166 tracts)
- `dc2_object_run2.2i_dr6_wfd_with_metacal`: static object + metacal catalog for Run 2.2i DR6 WFD (v1; 166 tracts)
```

yymao commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd_v1_with_metacal.yaml:
```diff
@@ -0,0 +1,37 @@
+    bands: 'griz'
+    apply_metacal_test3_fix: true
+    matching_method: MATCHING_FORMAT
+include_in_default_catalog_list: true
```
Hide the entry with `v1` designation from default view.
```suggestion
```

yymao commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd_v1.yaml:
```diff
@@ -0,0 +1,6 @@
+description: DC2 Run 2.2i DR6 WFD v1 Object Table
+creators: ['DESC DC2 Team']
+pixel_scale: 0.2
+include_in_default_catalog_list: true
```
Hide the entry with `v1` designation from default view.
```suggestion
```

yymao commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd.yaml:
```diff
@@ -0,0 +1,2 @@
+alias: dc2_object_run2.2i_dr6_wfd_v1
+
```
Show this entry in the default view.
```suggestion
include_in_default_catalog_list: true
```

yymao reviewed (changes requested):
Thanks, @heather999 --- I made some suggestions to make the catalog names without `v1` to be the user-facing entries. 

Should we also add `dc2_object_run2.2i_dr6_wfd_with_metacal.yaml` with the following?

```yaml
alias: dc2_object_run2.2i_dr6_wfd_v1_with_metacal
include_in_default_catalog_list: true
```

yymao reviewed (approved):
Thanks @heather999!
